frontend/packages/jobs/src/functions/processSavePullRequest.ts

```diff
@@ -213,13 +217,56 @@ export async function processSavePullRequest(
       )
     }
   }
+}
+
+export async function processSavePullRequest(
+  payload: SavePullRequestPayload,
+): Promise<SavePullRequestResult> {
+  const supabase = createClient()
+
+  const repository = await getRepositoryFromProjectId(
+    supabase,
+    payload.projectId,
+  )
+
+  const fileChanges = await getPullRequestFiles(
+    Number(repository.installationId),
+    repository.owner,
+    repository.name,
+    payload.prNumber,
+  )
+
+  const schemaPath = await getSchemaPathForProject(supabase, payload.projectId)
+
+  const prDetails = await getPullRequestDetails(
+    Number(repository.installationId),
+    repository.owner,
+    repository.name,
+    payload.prNumber,
+  )
+
+  const schemaFile = await fetchSchemaFileContent(
+    repository,
+    schemaPath,
+    prDetails.head.ref,
+  )
+
+  const fileChangesData = formatFileChangesData(fileChanges)
+
+  const prRecord = await getOrCreatePullRequestRecord(
+    supabase,
+    repository.id,
+    payload.prNumber,
+  )
+
+  await createOrUpdateMigrationRecord(supabase, prRecord.id, prDetails.title)
 
   return {
     success: true,
     prId: prRecord.id,
     repositoryId: repository.id,
-    schemaFiles,
+    schemaFile,
     fileChanges: fileChangesData,
-    branchName,
+    branchName: prDetails.head.ref,
   }
 }
```

Comment:
It's hard to tell because I've included a function split, but you can see the function name that processSavePullRequest is calling.